sui_programmability/examples/defi/sources/subscription.move

```diff
@@ -0,0 +1,103 @@
+/// Module implementing generic Subscription with limited number of uses.
+///
+/// This specific example shows how this model can be used within the AMM
+/// context to charge developers who make use of multiple shared objects
+/// which is expected to slow down all pools used.
+///
+/// The module itself makes use of HotPotato and Witness-based interface
+/// which can be implemented by other applications based on their needs.
+module defi::dev_pass {
+    use sui::tx_context::{TxContext};
+    use sui::object::{Self, Info};
+    use sui::transfer;
+
+    /// Owned object from which SingleUses are spawn
+    struct Subscription<phantom T> has key {
+        info: Info,
+        uses: u64
+    }
+
+    /// A single use potato.
+    struct SingleUse<phantom T> {}
+
+    /// Function to issue a membership with a number of uses.
+    public fun issue_subscription<T: drop>(_w: T, uses: u64, ctx: &mut TxContext): Subscription<T> {
+        Subscription {
+            info: object::new(ctx),
+            uses
```

Comment:
Do we care checking that uses != 0
?

Reply:
We could do, but since this is "implementable" function, we pass this responsibility to app developers.